containers/src/Data/Map/Internal.hs

```diff
@@ -3410,8 +3413,7 @@ instance (Ord k) => GHCExts.IsList (Map k v) where
 -- If the list contains more than one value for the same key, the last value
 -- for the key is retained.
 --
--- If the keys of the list are ordered, linear-time implementation is used,
--- with the performance equal to 'fromDistinctAscList'.
+-- If the keys of the list are ordered, a linear-time implementation is used.
```

Comment:
I made a rough attempt but it performed worse for the not-sorted case. I'll try to see if it can be improved, but it'd probably be best in a separate PR.